packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import styles from './visit-form.component.scss';
+import Button from 'carbon-components-react/es/components/Button';
+import DatePicker from 'carbon-components-react/es/components/DatePicker';
+import DatePickerInput from 'carbon-components-react/es/components/DatePickerInput';
+import Form from 'carbon-components-react/es/components/Form';
+import { useTranslation } from 'react-i18next';
+import { Column, Grid, Row } from 'carbon-components-react/es/components/Grid';
+import TimePicker from 'carbon-components-react/es/components/TimePicker';
+import TimePickerSelect from 'carbon-components-react/es/components/TimePickerSelect';
+import SelectItem from 'carbon-components-react/es/components/SelectItem';
+import ContentSwitcher from 'carbon-components-react/es/components/ContentSwitcher';
+import Switch from 'carbon-components-react/es/components/Switch';
+import Select from 'carbon-components-react/es/components/Select';
+import VisitTypeOverview from './visit-type-overview.component';
+import {
+  createErrorHandler,
+  detach,
+  getStartedVisit,
+  NewVisitPayload,
+  saveVisit,
+  showToast,
+  useLocations,
+  useSessionUser,
+  VisitMode,
+  VisitStatus,
+} from '@openmrs/esm-framework';
+import dayjs from 'dayjs';
+import { convertTime12to24, getDatePart } from './start-visit-helper';
+
+interface StartVisitFormProps {
+  isTablet: boolean;
+  patientUuid: string;
+}
+
+const StartVisitForm: React.FC<StartVisitFormProps> = ({ isTablet, patientUuid }) => {
+  const { t } = useTranslation();
+  const locations = useLocations();
+  const sessionUser = useSessionUser();
+  const isPM = useCallback(() => new Date().getHours() >= 12, []);
```

Comment:
Yes. Have a read through https://kentcdodds.com/blog/usememo-and-usecallback sometime, it's very helpful. You could also simply inline it below,

```typescript
const [timeFormat, setTimeFormat] = useState<string>(new Date().getHours() >=12 ? 'PM' : 'AM');
```

it's very simple and has basically no overhead. In fact, it simplifies the whole component since it's one less function I'm trying to keep track of when reading.